pkg/executor/docker/executor.go

```diff
@@ -90,282 +105,300 @@ func (e *Executor) ShouldBidBasedOnUsage(
 	return resource.NewChainedResourceBidStrategy().ShouldBidBasedOnUsage(ctx, request, usage)
 }
 
-//nolint:funlen,gocyclo // will clean up
-func (e *Executor) Run(
-	ctx context.Context,
-	request *executor.RunCommandRequest,
-) (*models.RunCommandResult, error) {
-	log.Ctx(ctx).Info().Msgf("running execution %s", request.ExecutionID)
-	ctx, cancel := context.WithCancel(ctx)
-	e.cancellers.Put(request.ExecutionID, cancel)
-	defer func() {
-		if cancelFn, found := e.cancellers.Get(request.ExecutionID); found {
-			e.cancellers.Delete(request.ExecutionID)
-			cancelFn()
+// Start initiates an execution based on the provided RunCommandRequest.
+// It performs several key tasks as part of the initiation:
+// 1. Checks if the execution is already started or completed and returns an error if so.
+// 2. Calls newDockerJobContainer to create and configure a Docker container for the execution.
+// 3. Registers a new executionHandler to manage the execution.
+// 4. Finally, it starts a new Goroutine to run the container.
+// It returns an error if any of the above steps fail.
+func (e *Executor) Start(ctx context.Context, request *executor.RunCommandRequest) error {
+	log.Ctx(ctx).Info().
+		Str("executionID", request.ExecutionID).
+		Str("jobID", request.JobID).
+		Msg("starting execution")
+
+	if handler, found := e.handlers.Get(request.ExecutionID); found {
+		if handler.active() {
+			return fmt.Errorf("starting execution (%s): %w", request.ExecutionID, executor.AlreadyStartedErr)
+		} else {
+			return fmt.Errorf("starting execution (%s): %w", request.ExecutionID, executor.AlreadyCompleteErr)
```

Comment:
I'm sure I've seen a linter (or gopls) complain before that errors should be `ErrSomething` instead of `SomethingErr`.

Reply:
I promise I ran with the precommit hook on every commit, I event edited these lines and re-ran the commit hook - no complaints. This either indicates we are missing a linter rule, or its misconfiguration. Regardless, I have renamed these vars to begin with `Err*`.